ecr-login/api/client.go

```diff
@@ -27,9 +28,27 @@ import (
 )
 
 const proxyEndpointScheme = "https://"
+const programName = "docker-credential-ecr-login"
+
+var ecrPattern = regexp.MustCompile(`(^[a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?`)
+
+func ExtractRegistryAndRegion(serverURL string) (string, string, error) {
+	if (strings.HasPrefix(serverURL, proxyEndpointScheme)) {
+		serverURL = strings.TrimPrefix(serverURL, proxyEndpointScheme)
+	}
+	matches := ecrPattern.FindStringSubmatch(serverURL)
+	if len(matches) == 0 {
+		return "", "", fmt.Errorf(programName + " can only be used with Amazon EC2 Container Registry.")
+	} else if len(matches) < 3 {
+		return "", "", fmt.Errorf(serverURL + "is not a valid repository URI for Amazon EC2 Container Registry.")
+	}
+	registry := matches[1]
+	region := matches[2]
+	return registry, region, nil
+}
 
 type Client interface {
-	GetCredentials(registry, image string) (*Auth, error)
+	GetCredentials(serverURL string) (*Auth, error)
```

Comment:
Since CLI code is still in PR, I can adapt the change of function names, but this will make it non-reusable for the CLI. The CLI only knows about the registryID when calling GetCredentials, and gets the region from aws config.